modules/collect/src/test/java/com/opengamma/strata/collect/result/PartialResultTest.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.collect.result;
+
+import static com.opengamma.strata.collect.TestHelper.assertSerialization;
+import static com.opengamma.strata.collect.TestHelper.coverBeanEquals;
+import static com.opengamma.strata.collect.TestHelper.coverImmutableBean;
+import static org.testng.Assert.assertEquals;
+
+import org.testng.annotations.Test;
+
+import com.google.common.collect.ImmutableList;
+
+/**
+ * Test {@link PartialResult}.
+ */
+@Test
+public class PartialResultTest {
```

Comment:
It seems that the callers need to decide ahead of time if they want the Result (Success/Failure) pattern or the PartialResult pattern.

If they choose the PartialResult path (as in the loading rows from a file example), then it appears that we expect that they will always get some form of valid result, and *may* get one or more errors.

They case I don't see and can imagine happening is where the caller is expected a PartialResult<T> and we don't have any valid result T. In this case, is the type T responsible for contain information as to whether it is valid or not?

So let's say we want to upload a portfolio of trades via:
```
PartialResult<MarginPortfolio> uploadTrades(List<TradeFile> trades);
```

What happens if all of the trades have errors and is now a FailureItem on the PartialResult? What if the MarginPortfolio does not support an empty set? Do we throw? Do we need to enforce that type T support some concept of failure or not being valid?